scipy/stats/_stats.pyx

```diff
@@ -513,6 +513,113 @@ cdef double _geninvgauss_pdf(double x, void *user_data) nogil except *:
+    cdef double inv_sqrt_2pi = 0.3989422804014327
+    return inv_sqrt_2pi * math.exp(-0.5 * z * z)
+
+
+cdef double _logphi(double z) nogil:
+    """evaluates the log of the normal PDF. Used in `studentized range`"""
+    cdef double log_inv_sqrt_2pi = -0.9189385332046727
+    return log_inv_sqrt_2pi - 0.5 * z * z
+
+
+cdef double _Phi(double z) nogil:
+    """evaluates the normal CDF. Used in `studentized range`"""
+    # use a custom evaluation because using cs.ndtr results in incorrect PDF
+    # evaluation at q=0 on 32bit systems
+    cdef double m_sqrt1_2 = 0.7071067811865475
+    return 0.5 * math.erfc(-z * m_sqrt1_2)
+
+
+cpdef double _genstudentized_range_cdf_logconst(double k, double df):
+    """Evaluates non-integrated terms in the cdf integrand as a log"""
+    cdef double log_2 = 0.6931471805599453
+    return (math.log(k) + (df / 2) * math.log(df)
+     - (math.lgamma(df / 2) + (df / 2 - 1) * log_2))
+
+
+cpdef double _genstudentized_range_pdf_logconst(double k, double df):
+    """Evaluates non-integrated terms in the pdf integrand as a log"""
+    cdef double log_2 = 0.6931471805599453
+    return (math.log(k) + math.log(k - 1) + (df / 2) * math.log(df)
+                 - (math.lgamma(df / 2) + (df / 2 - 1) * log_2))
+
+
+cdef double _genstudentized_range_cdf(int n, double[2] x, void *user_data) nogil:
+    # evaluates the integrand of Equation (3) by Batista, et al [2]
+    # destined to be used in a LowLevelCallable
+    q = (<double *> user_data)[0]
+    k = (<double *> user_data)[1]
+    df = (<double *> user_data)[2]
+    log_const = (<double *> user_data)[3]
+
+    s = x[1]
+    z = x[0]
+
+    # suitable terms are evaluated within logarithms to avoid under/overflows
+    log_terms = (log_const
+                 + (df - 1) * math.log(s)
+                 - (df * s * s / 2)
+                 + _logphi(z))
+
+    # multiply remaining term outside of log because it can be 0
+    return math.exp(log_terms) * math.pow(_Phi(z + q * s) - _Phi(z), k - 1)
+
+
+cdef double _genstudentized_range_cdf_asymptotic(double z, void *user_data) nogil:
+    # evaluates the integrand of equation (2) by Lund, Lund, page 205. [4]
+    # destined to be used in a LowLevelCallable
+    q = (<double *> user_data)[0]
+    k = (<double *> user_data)[1]
+
+    return k * _phi(z) * math.pow(_Phi(z + q) - _Phi(z), k - 1)
+
+
+cdef double _genstudentized_range_pdf(int n, double[2] x, void *user_data) nogil:
+    # evaluates the integrand of equation (4) by Batista, et al [2]
+    # destined to be used in a LowLevelCallable
+    q = (<double *> user_data)[0]
+    k = (<double *> user_data)[1]
+    df = (<double *> user_data)[2]
+    log_pdf_const = (<double *> user_data)[3]
+
+    z = x[0]
+    s = x[1]
+
+    # suitable terms are evaluated within logarithms to avoid under/overflows
+    log_terms = (log_pdf_const
+                 + (df - 1) * math.log(s)
+                 - df * s * s / 2
+                 + math.log(s)
```

Comment:
```suggestion
                 + df * math.log(s)
                 - df * s * s / 2
```
Right?

Reply:
Looks like you agree, so I think it would be a good simplification to implement ; )

Reply:
Coming in the next commit